src/wp-includes/post.php

```diff
@@ -7643,11 +7643,11 @@ function _publish_post_hook( $post_id ) {
  *
  * @since 3.1.0
  *
- * @param int|WP_Post $post Post ID or post object.
+ * @param int|WP_Post|null $post Optional. Post ID or post object. Default null.
```

Comment:
I think we may want to expand on what `null` means here with some text similar to the parameter description from `get_post`.

Reply:
Agree with Timothy, aside from `get_post()` which goes in to full details, the most common text appears to be:

```
 * @param WP_Post|int|null $post   Optional. Post ID or post object. Defaults to global $post.
```